### PR TITLE
ADR: Extract Reporting

### PR DESCRIPTION
**Pull request by peteraba**
As agreed, here I'm adding an ADR about reporting extracted from the monolith.

### CODE REVIEW

File: adr-extract-reporting.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+crashes. We think it’s mostly due to spikes in usage, and the number of
+customers using the system.
+
+Change is difficult and risky in this large monolith - whenever a change is
+made, it takes too long and something else usually breaks.
+
+To tackle both issues, we decided to first extract self-containing components
+from the main system and enable asynchronous handling of tasks where possible.
+
+As a first step we will start with components which appear to be easy to isolate
```
Comment (anatolfernandez) [inline adr-extract-reporting.md]:
```suggestion
As a first step, we will start with components which appear to be easy to isolate
```

File: adr-extract-reporting.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+7. Refactor any existing code which triggers reporting processes to use the new
+reporting queues to initiate their tasks.
+
+## Consequences
+
+1. Load on our system during report generation drops without much development
+effort.
+2. Scalable reporting which can be adjusted to our budget and priorities. (In
+case of budget limitations, we can also introduce queues based on priorities.)
+3. Further refactoring, or complete rewriting of the reporting module is now
```
Comment (anatolfernandez) [inline adr-extract-reporting.md:51]:
```suggestion
3. Further refactoring, or complete rewriting of the reporting module, is now 
```

File: adr-extract-reporting.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# ADR 1: Extract reporting from the main system (a.k.a the Monolith)
```
Comment (anatolfernandez) [inline adr-extract-reporting.md]: I'd capitalize the title here:

```suggestion
# ADR 1: Extract Reporting from the Main System (a.k.a the Monolith)
```

**Review by anatolfernandez — APPROVED** [review]
LGTM, just a couple of minor comments!

File: adr-extract-reporting.md (hunk 2 above)
Comment (peteraba) [reply adr-extract-reporting.md:51]: I think it was correct as it was.

**Pull request merged**